examples/mxnet/gat/gat_batch.py

```diff
@@ -168,39 +187,55 @@ def main(args):
                 args.attn_drop,
                 args.residual)
 
-    model.initialize(ctx=ctx)
+    model.initialize()
+
+    if cuda:
+        model.collect_params().reset_ctx(ctx)
 
     # use optimizer
     trainer = gluon.Trainer(model.collect_params(), 'adam', {'learning_rate': args.lr})
 
-    # initialize graph
     dur = []
     for epoch in range(args.epochs):
         if epoch >= 3:
             t0 = time.time()
         # forward
         with mx.autograd.record():
             logits = model(features)
-            loss = mx.nd.softmax_cross_entropy(logits, labels)
-
-        #optimizer.zero_grad()
-        loss.backward()
-        trainer.step(features.shape[0])
-        loss.wait_to_read()
+            loss = mx.nd.softmax_cross_entropy(logits[mask].squeeze(), labels[mask].squeeze())
+            loss.backward()
+        trainer.step(mask.shape[0])
 
         if epoch >= 3:
             dur.append(time.time() - t0)
-            print("Epoch {:05d} | Loss {:.4f} | Time(s) {:.4f} | ETputs(KTEPS) {:.2f}".format(
-                epoch, loss.asnumpy()[0], np.mean(dur), n_edges / np.mean(dur) / 1000))
+        if epoch % 100 == 0:
+            val = logits[val_mask]
+            val = val.asnumpy().squeeze()
+            val_labels = labels[val_mask].asnumpy().squeeze()
+            max_index = np.argmax(val, axis=1)
+            accuracy = np.sum(np.where(max_index == val_labels, 1, 0))/len(val_labels)
+            print("Validation accuracy: {:.2f}".format(accuracy))
```

Comment:
Move evaluation code into a separate function so that it can be shared by validation and testing